src/Maestro/Types/V1/Blocks.hs

```diff
@@ -0,0 +1,60 @@
+-- | Module to define types for /\"Blocks\"/ category endpoints defined at [docs.gomaestro.org](https://docs.gomaestro.org/docs/category/blocks).
+
+module Maestro.Types.V1.Blocks
+  ( BlockDetails (..)
+  , TimestampedBlockDetails (..)
+  ) where
+
+import           Data.Text               (Text)
+import           Data.Word               (Word64, Word32)
+import           Deriving.Aeson
+import           Maestro.Types.Common
+import           Maestro.Types.V1.Pools   (PoolId)
+import           Maestro.Types.V1.Common
+
+-- | Complete block Details when queried by its hash or height.
+data BlockDetails = BlockDetails
+  { _blockDetailsHash :: !BlockHash
+  -- ^ Block hash (identifier)
+  , _blockDetailsHeight :: !BlockHeight
+  -- ^ Block height
+  , _blockDetailsAbsoluteSlot :: !Word64
+  -- ^ Absolute slot of the block which includes the transaction
+  , _blockDetailsBlockProducer :: !(Maybe (Bech32StringOf PoolId))
+  -- ^ Block producer identifier name
+  , _blockDetailsConfirmations :: !Word64
+  -- ^ Total number of confirmations
+  , _blockDetailsEpoch :: !EpochNo
+    -- ^ Epoch number
+  , _blockDetailsEpochSlot :: !EpochSize
+    -- ^ Epoch Slot number
+  , _blockDetailsPreviousBlock :: !(Maybe BlockHash)
+    -- ^ Previous block hash
+  , _blockDetailsScriptInvocations :: !Word32
+    -- ^ Previous block hash
+  , _blockDetailsSize :: !Word32
+    -- ^ Block total size
+  , _blockDetailsTimestamp :: !Text
```

Comment:
Do we need `Text` here? Can't we use similar types we used earlier for time like [here](https://github.com/maestro-org/haskell-sdk/blob/deb1e133cd7de989e9e85c2b44e9361f52e93776/src/Maestro/Types/V1/General.hs#L110).

Reply:
Nice suggestion, I was taking a look to the maestro response, and we can't use for instance `NominalDiffTime` because we get the following error while decoding
```
ServantClientError (DecodeFailure "Error in $.data.timestamp: parsing NominalDiffTime failed, expected Number, but encountered String"
```
but we can use `LocalTime`.